src/ray/raylet/agent_manager.cc

```diff
@@ -162,10 +142,11 @@ void AgentManager::CreateRuntimeEnv(
 
   if (runtime_env_agent_client_ == nullptr) {
     // If the agent cannot be restarted anymore, fail the request.
-    if (agent_restart_count_ >= RayConfig::instance().agent_max_restart_count()) {
+    if (disable_agent_client_ || agent_failed_) {
```

Comment:
If agent_failed == true, this line should never reach? Maybe instead

```
RAY_CHECK(!agent_failed)
if (disable_agent_client_)
```

Reply:
Rmove the flag `agent_failed_`.